core/variant/array.cpp

```diff
@@ -95,22 +95,22 @@ Array::ConstIterator Array::end() const {
 	return ConstIterator(_p->array.ptr() + _p->array.size(), _p->read_only);
 }
 
-Variant &Array::operator[](int p_idx) {
-	ERR_FAIL_INDEX_MSG(p_idx, size(), _p->array[0]);
-	if (unlikely(_p->read_only)) {
-		*_p->read_only = _p->array[p_idx];
-		return *_p->read_only;
-	}
-	return _p->array.write[p_idx];
+Variant Array::operator[](int p_idx) {
+    ERR_FAIL_INDEX_V_MSG(p_idx < 0 || p_idx >= size(), Variant());
```

Comment:
Please make sure you set your code editor to use tabs for indentation. I highly recommend using an IDE to edit the code, because an IDE will adjust the setting automatically for you, depending on existing indentation in the respective file.